utils/riff/recommendations/index.jsx

```diff
@@ -0,0 +1,237 @@
+ * @returns An array  of Recommendation objects representing all the Recommendations that might be displayed.
+ *
+ * TODO: Recommendations' definitions could possibly be moved to an external file or database.
+ */
+function getRecommendations(userId, teamId, learningGroups, courseStartTime) {
+    const plgChannel = getGroupChannel(userId, learningGroups, 'plg');
+    const capstoneChannel = getGroupChannel(userId, learningGroups, 'capstone');
+
+    const townSquareId = internal.getTownSquareId(teamId);
+    const hasPostedInTownSquare = new Recommendation(
+        userId,
+        1, // first week to display this recommendation
+        2, // number of weeks to display for
+        0, // relative priority to other recs for this week
+        () => didUserPostInChannel(userId, townSquareId),
+        React.createElement('span', null, [
+            React.createElement('span', null, 'Introduce yourself in '),
+            <Link
+                key={1}
+                to={internal.getChannelURL(townSquareId)}
+            >
+                {'Town Square'}
+            </Link>,
+            React.createElement('span', null, ' to begin connecting with other learners in the course.'),
+        ]),
+        courseStartTime,
+        'User has posted in Town Square ever.',
+    );
+
+    const hasTwoMeetingsPerWeek = new Recommendation(
+        userId,
+        2, // first week to display this recommendation
+        // eslint-disable-next-line no-magic-numbers
+        7, // number of weeks to display for
+        0, // relative priority to other recs for this week
+        async () => {
+            const currentWeek = getCurrentWeekNumber(courseStartTime);
+            let numMtgsPLG = 0;
+            let numMtgsCap = 0;
+            if ('teammateIds' in plgChannel) {
+                numMtgsPLG = await numberOfUserMeetingsDuringWeeks(userId, plgChannel.teammateIds, currentWeek, 1, courseStartTime);
+            }
+            if ('teammateIds' in capstoneChannel) {
+                numMtgsCap = await numberOfUserMeetingsDuringWeeks(userId, capstoneChannel.teammateIds, currentWeek, 1, courseStartTime);
+            }
+            return numMtgsPLG + numMtgsCap >= 2;
+        },
+        'Have two Riff calls with your team this week. Teams who do are more likely to finish the course and have higher grades.',
+        courseStartTime,
+        'User has held two Riff meetings with their PLG this week.',
+    );
+
+    const hasConnectedWithCourse = new Recommendation(
+        userId,
+        // eslint-disable-next-line no-magic-numbers
+        5, //first week to display this recommendation
+        3, // number of weeks to display for
+        0, // relative priority to other recs for this week
+        async () => {
+            const userConnections = UserStore.getUserAnalytics()[MMUserAnalyticsDatasets.USER_INTERACTIONS].filter((interaction) => {
```

Comment:
Do we need to change this to use the updated method for getting user analytics?

Reply:
If you connected the component index to the store, you could just pass this info here:
```
const allRecs = getRecommendations(
                this.props.currentUserId,
                this.props.currentTeamId,
                this.props.userLearningGroups,
                startTime,
            );
```